plugin/etcd/plugin.go

```diff
@@ -0,0 +1,451 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"encoding/base64"
+	"io"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/coreos/etcd/pkg/transport"
+	fmt "github.com/jhunt/go-ansi"
+	"go.etcd.io/etcd/clientv3"
+
+	"github.com/starkandwayne/shield/plugin"
+)
+
+func main() {
+	p := EtcdPlugin{
+		Name:    "Etcd Backup Plugin",
+		Author:  "Jason Zhou, Pururva Lakkad, Naveed Ahmad, Sriniketh Varma Dasarraju",
+		Version: "0.0.1",
+		Features: plugin.PluginFeatures{
+			Target: "yes",
+			Store:  "no",
+		},
+		Example: `
+			{
+			"endpoints" : "https://192.168.42.45:2379"   # REQUIRED
+
+			"auth"  : false												# is role based or cert based auth enabled on the etcd cluster
+			"user"  : "admin",            								# username for role based authentication
+			"pass"  : "p@ssw0rd"          								# password for role based authentication
+			"clientCertPath" : "/tmp/test-certs/test-client.crt" 		# path to client certificate
+			"clientKeyPath"  : "/tmp/test-certs/test-client-key.key" 	# path to client key
+			"caCertPath"     : "/tmp/test-certs/test-ca.crt" 			# path to CA certificate
+			"fullOverwrite"  : "false"								    # enable or disable full overwrite of the cluster
+			"prefix"		 : "starkandwayne/"							# backup specific keys
```

Comment:
Check the help output of the plugin, and adjust white space as necessary to make it more aesthetically pleasing.  Replace tabs with spaces while you're at it.